app/src/common/shared/com/igalia/wolvic/ui/widgets/prompts/ColorPromptWidget.java

```diff
@@ -0,0 +1,79 @@
+package com.igalia.wolvic.ui.widgets.prompts;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.widget.Button;
+
+import androidx.annotation.NonNull;
+
+import com.igalia.wolvic.R;
+import com.igalia.wolvic.audio.AudioEngine;
+import com.skydoves.colorpickerview.ColorPickerView;
+import com.skydoves.colorpickerview.flag.BubbleFlag;
+import com.skydoves.colorpickerview.flag.FlagMode;
+import com.skydoves.colorpickerview.preference.ColorPickerPreferenceManager;
+
+public class ColorPromptWidget extends PromptWidget {
+
+    private AudioEngine mAudio;
+    private Button mCancelButton, mOkButton;
+
+    public ColorPromptWidget(Context aContext) {
+        super(aContext);
+        initialize(aContext);
+    }
+
+    public ColorPromptWidget(Context aContext, AttributeSet aAttrs) {
+        super(aContext, aAttrs);
+        initialize(aContext);
+    }
+
+    public ColorPromptWidget(Context aContext, AttributeSet aAttrs, int aDefStyle) {
+        super(aContext, aAttrs, aDefStyle);
+        initialize(aContext);
+    }
+
+    protected void initialize(Context aContext) {
+        inflate(aContext, R.layout.prompt_color, this);
+        new ColorPickerView.Builder(getContext()).build();
+
+        ColorPickerView colorPickerView = findViewById(R.id.colorPickerView);
+        BubbleFlag bubbleFlag = new BubbleFlag(getContext());
+        bubbleFlag.setFlagMode(FlagMode.ALWAYS);
+        colorPickerView.setFlagView(bubbleFlag);
+        ColorPickerPreferenceManager manager = ColorPickerPreferenceManager.getInstance(getContext());
+
+        mAudio = AudioEngine.fromContext(aContext);
+        mLayout = findViewById(R.id.layout);
+        mTitle = findViewById(R.id.title);
+        mCancelButton = findViewById(R.id.cancelButton);
+        mOkButton = findViewById(R.id.okButton);
+
+        mCancelButton.setOnClickListener(v -> {
+            if (mAudio != null) {
+                mAudio.playSound(AudioEngine.Sound.CLICK);
+            }
+            if (mPromptDelegate != null && mPromptDelegate instanceof ColorPromptDelegate) {
+                mPromptDelegate.dismiss();
+                manager.clearSavedAllData();
+            }
+            hide(REMOVE_WIDGET);
+        });
+
+        mOkButton.setOnClickListener(v -> {
+            if (mAudio != null) {
+                mAudio.playSound(AudioEngine.Sound.CLICK);
+            }
+            if (mPromptDelegate != null && mPromptDelegate instanceof ColorPromptDelegate) {
+                String colorCodeHex = "#" + colorPickerView.getColorEnvelope().getHexCode().substring(2, 8);
```

Comment:
Could you add a comment here explaining what this code does?

Reply:
Added the same.